adafruit_register/register_accessor.py

```diff
@@ -0,0 +1,165 @@
+          Buffer must be long enough to be read all data sent by the device.
+        :return: None
+        """
+
+        self._pack_address_into_buffer(address)
+        self._shift_rw_cmd_bit_into_first_byte(1)
+        with self.spi_device as spi:
+            spi.write(self.address_buffer)
+            spi.readinto(buffer)
+
+    def write_register(
+        self,
+        address: int,
+        buffer: bytearray,
+    ):
+        """
+        Write register value over SPIDevice.
+
+        :param int address: The register address to read.
+        :param bytearray buffer: Buffer that will be written to the register.
+        :return: None
+        """
+        self._pack_address_into_buffer(address)
+        self._shift_rw_cmd_bit_into_first_byte(0)
+        with self.spi_device as spi:
+            spi.write(self.address_buffer)
+            spi.write(buffer)
+
+
+class I2CRegisterAccessor(RegisterAccessor):
+    """
+    RegisterAccessor class for I2C bus transport. Provides interface to read/write
+    registers over I2C
+
+    :param I2CDevice i2c_device: I2C device to communicate over
+    :param int address_width: The number of bytes in the address
+    """
+
+    def __init__(self, i2c_device: I2CDevice, address_width: int = 1, lsb_first=True):
+        super().__init__(address_width, lsb_first)
+        self.i2c_device = i2c_device
+
+        # buffer that will hold address + data for write operations, will grow as needed
+        self._full_buffer = bytearray(address_width + 1)
+
+    def read_register(self, address: int, buffer: bytearray):
+        """
+        Read register value over I2CDevice.
+
+        :param int address: The register address to read.
+        :param bytearray buffer: Buffer that will be used to read register data into.
+          Buffer must be long enough to be read all data sent by the device.
+        :return: None
+        """
+
+        self._pack_address_into_buffer(address)
+        with self.i2c_device as i2c:
+            i2c.write_then_readinto(self.address_buffer, buffer)
+
+    def write_register(self, address: int, buffer: bytearray):
+        """
+        Write register value over I2CDevice.
+
+        :param int address: The register address to read.
+        :param bytearray buffer: Buffer of data that will be written to the register.
+
+        :return: None
+        """
+        # grow full buffer if needed
+        if self.address_width + len(buffer) > len(self._full_buffer):
+            self._full_buffer = bytearray(self.address_width + len(buffer))
+
+        # put address into full buffer
+        self._pack_address_into_buffer(address)
+        for i in range(self.address_width):
+            self._full_buffer[i] = self.address_buffer[i]
+
+        # put data into full buffer
+        for i, b in enumerate(buffer):
+            self._full_buffer[i + self.address_width] = b
+
+        with self.i2c_device as i2c:
+            i2c.write(self._full_buffer)
```

Comment:
The full buffer may be longer than we need so we need to set end.

```suggestion
            i2c.write(self._full_buffer, end=self.address_width+len(buffer))
```